src/server/journal/types.h

```diff
@@ -11,37 +14,59 @@ namespace journal {
 
 enum class Op : uint8_t {
   NOOP = 0,
-  LOCK = 1,
-  UNLOCK = 2,
-  LOCK_SHARD = 3,
-  UNLOCK_SHARD = 4,
-  SCHED = 5,
-  VAL = 10,
-  DEL,
-  MSET,
+  SELECT = 1,
+  COMMAND = 2,
 };
 
 // TODO: to pass all the attributes like ttl, stickiness etc.
 struct Entry {
-  Entry(Op op, DbIndex did, TxId tid, std::string_view skey)
-      : opcode(op), db_ind(did), txid(tid), key(skey) {
+  using Payload = std::variant<std::monostate,
+                               CmdArgList,                            // Full single key command.
+                               std::pair<std::string_view, ArgSlice>  // Command + shard parts.
+                               >;
+
```

Comment:
In theory, if we had `Span<char>` and `const Span<const char>`, then we could safely interchange them even though templates are invariant, because we don't allow mutating the ArgSlice in any way, which means that we can't break const guarantees even if we kept the CmdArgSlice (i.e. the vector<T> into vector<const T> problem)

Reply:
However we don't want to give up on the whole string_view convenience

We could also implement some wrapping container that indexes into different types depending on constness, but it'd be really really tricky...

Reply:
You are over-optimizing. It's not worth your time.  Connect the dots and then we can optimize/tune every part.